verifier/testreport.py

```diff
@@ -572,6 +572,15 @@ def characterize_failures_by_options(self, failing_tests):
                     except:
                         continue
 
+                for key in ['language_label', 'ignorePunctuation', 'compare_result', 'compare_type', 'test_description']:
+                    if test.get(key):  # For collation results
+                        value = test[key]
+                        if key not in results:
+                            results[key] = {}
+                        if value in results[key]:
+                            results[key][value].append(label)
+                        else:
+                            results[key][value] = [label]
```

Comment:
optional: if you initialize `results` using [Python's support for auto-initializing the values in a nested dictionary](https://stackoverflow.com/questions/651794/whats-the-best-way-to-initialize-a-dict-of-dicts-in-python), such as:

```
results = collections.defaultdict(lambda : collections.defaultdict(list))
```

...then the body of this block of code would become a single line.

```suggestion
                for key in ['language_label', 'ignorePunctuation', 'compare_result', 'compare_type', 'test_description']:
                    results[key][value].append(label)
```

Reply:
Thx. Will resolve later.